### PR TITLE
Make protocol level a run time option

### PR DESCRIPTION
**Pull request by embhorn**
Allows broker examples that do not work with MQTTv5 to run even when `--enable-mqtt5` option is used.

Also fixed an include issue for simple example without TLS enabled.

### CODE REVIEW

File: examples/aws/awsiot.c [hunk 1]
```diff
@@ -27,15 +27,18 @@
...
  *
  *  The "wc_GetTime" API was added in 3.9.1 and if not present you'll need to implement
  *  your own version of this to get current UTC seconds or update your wolfSSL library
 */
 
-/* This example requires features in wolfSSL 3.9.1 or later */
-#if defined(ENABLE_MQTT_TLS)
+/* This example requires features in wolfSSL 3.9.1 or later and
+ * will not work with MQTTv5 */
+#if defined(ENABLE_MQTT_TLS) && !defined(WOLFMQTT_V5)
```
Comment (dgarske) [inline examples/aws/awsiot.c]: With the run-time option we should still be able to use this example with `WOLFMQTT_V5` defined. I think you can revert this change.

File: examples/azure/azureiothub.c [hunk 2]
```diff
@@ -38,8 +40,9 @@
  *  your own version of this to get current UTC seconds or update your wolfSSL library
 */
 
-/* This example requires features in wolfSSL 3.9.1 or later */
-#if defined(ENABLE_MQTT_TLS)
+/* This example requires features in wolfSSL 3.9.1 or later and
+ * will not work with MQTTv5 */
+#if defined(ENABLE_MQTT_TLS) && !defined(WOLFMQTT_V5)
```
Comment (dgarske) [inline examples/azure/azureiothub.c]: With the run-time option we should still be able to use this example with `WOLFMQTT_V5` defined. I think you can revert this change.

**Comment by embhorn** [comment]
> ... make `protocol_level` always be present (not just for v5)? Then set it to default to v3.1.1 normally and default to v5 when `WOLFMQTT_V5` is defined.

The information from protocol level is only relevant if there is more than one protocol, which is only possible when `WOLFMQTT_V5` is defined. It already does the change the default when V5 is enabled. That's also why the protocol level has to be set explicitly in the AWS and Azure examples when V5 is enabled.
